### PR TITLE
PKG Add python-solvespace

### PR DESCRIPTION
**Pull request by fuyb1992**
### Description

python-sovlespace is a python library from the geometric constraints solver of SolveSpace, an open source CAD software.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by fuyb1992** [comment]
Need for help! CI failed at test-packages-chrome when it try to import scikit-image, but this PR changes nothing about scikit-image

.

**Comment by ryanking13** [comment]
Timeout happens occasionally in CI, it's not your fault. I re-triggered CI.

File: packages/python_solvespace/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package:
+  name: python_solvespace
+  version: 3.0.6
+source:
+  url: https://files.pythonhosted.org/packages/50/a6/4e01d2c90de41a431041f813aebada874d05d6d578ce25b2462872d2a3e6/python_solvespace-3.0.6.tar.gz
+  sha256: 951b4cf3858dc337bbfa9f6d9efe9fd6e8dfa7fed022556196add1ba5a56b045
+  extras:
+    - [Eigen/Cholesky, python_solvespace/include/Eigen/Cholesky]
```
Comment (hoodmane) [inline packages/python_solvespace/meta.yaml:8]: Please get rid of all of these extras. We will not vendor Eigen into Pyodide. I think you should be able to replace it with:
```yaml
build:
   script: |
      git clone https://gitlab.com/libeigen/eigen.git --depth 1 --branch 3.4
      ln -s eigen/Eigen python_solvespace/include/
```

**Review by hoodmane — COMMENTED** [review]
Thanks for this PR @fuyb1992. Please add a `test_python_solvespace.py` file. You can use `run_in_pyodide` and some subset of the solvespace unit tests. In particular, it's good to try to cover the most important C/C++ functions at least once to make sure that they are linked correctly. Also, please unvendor Eigen.

File: packages/python_solvespace/meta.yaml (hunk 1 above)
Comment (fuyb1992) [reply packages/python_solvespace/meta.yaml]: Thanks a lot, the build script works, I remove all extras in `meta.yaml` and add `test_python_solvespace.py` in https://github.com/pyodide/pyodide/pull/2287/commits/6177a5c96982f1b207f7b4b44b933a9b070b13bb. I had reported the missing file issue to KmolYuan,  he probably fix the issue in the next version of python-solvespace.

**Review by hoodmane — APPROVED** [review]

File: packages/python_solvespace/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package:
+  name: python_solvespace
+  version: 3.0.6
+source:
+  url: https://files.pythonhosted.org/packages/50/a6/4e01d2c90de41a431041f813aebada874d05d6d578ce25b2462872d2a3e6/python_solvespace-3.0.6.tar.gz
+  sha256: 951b4cf3858dc337bbfa9f6d9efe9fd6e8dfa7fed022556196add1ba5a56b045
+build:
+  script: |
+    git clone https://gitlab.com/libeigen/eigen.git --depth 1 --branch 3.4
```
[the remark this answers is not in the document]
Comment (fuyb1992) [reply packages/python_solvespace/meta.yaml]: I change the scrpit from git to wget, and it works fine.  I don't know how get Eigen from another a separate package as @hoodmane described，probably using wget in build script is more easier to understand for people new to pyodide.

**Comment by hoodmane** [comment]
Thanks @fuyb1992!

**Pull request merged**